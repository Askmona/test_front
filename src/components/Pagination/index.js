import React from 'react';
import styled from 'styled-components';
import { mediaQueries } from '../../theme/index.js';
import PropTypes from 'prop-types';

const PaginationWrapper = styled.div`
  display: flex;
  justify-content: center;
  margin: 2rem;
  text-align: center;
`;

const LinkPaginate = styled.a`
  color: #4054b2;
  font-weight: bold;
  font-size: 1.2em;
  cursor: pointer;
  padding: 1rem
  ${mediaQueries('s')`
    font-size: 1em
  `};
`;

const CurrentNum = styled.a`
  padding: 0px;
  width: 50px;
  height: 50px;
  line-height: 50px;
  border-radius: 50%;
  background: #4054b2;
  color: #FFF;
  font-size: 1.2em;
  transition: .3s;
  font-weight: bold;
  ${mediaQueries('s')`
    font-size: 1em;
  `};
  ${mediaQueries('xs')`
    padding: 0rem 1rem;
  `};
`;

const Pagination = ({
  currentPage,
  pageTotal,
  handleClickPrev,
  handleClickNext,
  handleClickLast,
  handleClickFirst,
}) => {
  const page = Math.ceil(currentPage / 8 + 1);
  const lastPage = Math.ceil(pageTotal / 8 + 1);
  if ( pageTotal <= 8 ) {
    return null;
  }
  return (
    <PaginationWrapper>
      <LinkPaginate
        data-jest='paginate-first'
        onClick={handleClickFirst}
      >
        {page !== 1 ? '1' : ''}
      </LinkPaginate>
      <LinkPaginate data-jest='paginate-prev' onClick={handleClickPrev}>
        {page !== 1 ? '<' : ''}
      </LinkPaginate>
      {page >= 2 &&
      <LinkPaginate data-jest='paginate-prev2' onClick={handleClickPrev}>
        {page - 1}
      </LinkPaginate>}
      <CurrentNum data-jest='paginate-current'>
        {page}
      </CurrentNum>
      {!(page >= lastPage) &&
        <LinkPaginate data-jest='paginate-next' onClick={handleClickNext}>
          {page + 1}
        </LinkPaginate>}
      <LinkPaginate data-jest='paginate-next2' onClick={handleClickNext}>
        {page !== lastPage ? '>' : ''}
      </LinkPaginate>
      <LinkPaginate data-jest='paginate-last' onClick={handleClickLast}>
        {page !== lastPage ? lastPage : ''}
      </LinkPaginate>
    </PaginationWrapper>
  );
}

Pagination.propTypes = {
  currentPage: PropTypes.number.isRequired,
  pageTotal: PropTypes.number.isRequired,
  handleClickPrev: PropTypes.func.isRequired,
  handleClickNext: PropTypes.func.isRequired,
  handleClickLast: PropTypes.func.isRequired,
  handleClickFirst: PropTypes.func.isRequired,
}

export default Pagination;
