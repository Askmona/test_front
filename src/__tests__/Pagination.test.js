import React from 'react';
import { shallow, mount } from 'enzyme';
import Pagination from '../components/Pagination';

describe('<Pagination />', () => {
  const onHandleNextMock = jest.fn();
  const onHandlePrevMock = jest.fn();
  const onHandleLastMock = jest.fn();
  const onHandleFirstMock = jest.fn();

  const props = {
    pageTotal: 1253,
    currentPage: 8,
    handleClickNext: () => onHandleNextMock(),
    handleClickPrev: () => onHandlePrevMock(),
    handleClickLast: () => onHandleLastMock(),
    handleClickFirst: () => onHandleFirstMock(),
  }
  const wrapper = shallow(<Pagination {...props} />)
  it('Shoulds render.', () => {
    expect(mount(<Pagination {...props} />)).toMatchSnapshot();
  })
  it('Should call "handleClickFirst" when clicking on the first paginate.', () => {
    wrapper.find("[data-jest='paginate-first']").simulate('click');
    expect(onHandleFirstMock).toHaveBeenCalled();
  });

  it('Should call "handleClickPrev" when clicking on the prev paginate.', () => {
    wrapper.find("[data-jest='paginate-prev']").simulate('click');
    expect(onHandlePrevMock).toHaveBeenCalled();
  });

  it('Should call "handleClickPrev" when clicking on the second prev paginate.', () => {
    wrapper.find("[data-jest='paginate-prev2']").simulate('click');
    expect(onHandlePrevMock).toHaveBeenCalled();
  });

  it('Should call "handleClickNext" when clicking on the next paginate.', () => {
    wrapper.find("[data-jest='paginate-next']").simulate('click');
    expect(onHandleNextMock).toHaveBeenCalled();
  });

  it('Should call "handleClickNext" when clicking on the second next paginate.', () => {
    wrapper.find("[data-jest='paginate-next2']").simulate('click');
    expect(onHandleNextMock).toHaveBeenCalled();
  });

  it('Should call "handleClickLast" when clicking on the last paginate.', () => {
    wrapper.find("[data-jest='paginate-last']").simulate('click');
    expect(onHandleLastMock).toHaveBeenCalled();
  });
});
