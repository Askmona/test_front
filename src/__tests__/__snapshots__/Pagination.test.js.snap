// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Pagination /> Shoulds render. 1`] = `ReactWrapper {}`;
